aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,312 @@
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/OrderingController.cs
+  :language: c#
+  :emphasize-lines: 1,10,18,24
+  :lines: 8-34
+  :dedent: 4
+
+The resulting order is:
+	1. The Controller ``OnActionExecuting``
+	2. The Global filter ``OnActionExecuting``
+	3. The Class filter ``OnActionExecuting``
+	4. The Method filter ``OnActionExecuting``
+	5. The Method filter ``OnActionExecuted``
+	6. The Class filter ``OnActionExecuted``
+	7. The Global filter ``OnActionExecuted``
+	8. The Controller ``OnActionExecuted``
+
+.. note:: ``IOrderedFilter`` trumps scope when determining the order in which filters will run. Filters are sorted first by order, then scope is used to break ties. Order defaults to 0 if not set.
+
+To modify the default, scope-based order, you could explicitly set the ``Order`` property of the class-level and/or method-level filters. For example, ``[OrderLoggingActionFilter(Name = "Method Level Attribute", Order=-1)]``. In this case, a value of less than zero would ensure this filter ran before both the Global and Class level filters.
+
+The new order would be:
+	#. The Controller ``OnActionExecuting``
+	#. The Method filter ``OnActionExecuting``
+	#. The Global filter ``OnActionExecuting``
+	#. The Class filter ``OnActionExecuting``
+	#. The Class filter ``OnActionExecuted``
+	#. The Global filter ``OnActionExecuted``
+	#. The Method filter ``OnActionExecuted``
+	#. The Controller ``OnActionExecuted``
+
+.. _authorization-filters:
+
+Authorization Filters
+---------------------
+*Authorization Filters* control access to action methods, and are the first filters to be executed within the filter pipeline. They have only a before stage, unlike most filters that support before and after methods. You should only write a custom authorization filter if you are writing your own authorization framework. Note that you should not throw exceptions within authorization filters, since nothing will handle the exception (exception filters won't handle them). Instead, issue a challenge or find another way. They are covered in the :doc:`Security </security/index>` section of the documentation.
+
+Learn more about :doc:`/security/authorization/authorization-filters`.
+
+.. _resource-filters:
+
+Resource Filters
+----------------
+*Resource Filters* implement either the ``IResourceFilter`` or ``IAsyncResourceFilter`` interface, and their execution wraps most of the filter pipeline (only :ref:`authorization-filters` run before them). Resource filters are expecially useful if you need to short-circuit most of the work a request is doing. Caching would be one example use case for a resource filter, since if the response is already in the cache, the filter can immediately set a result and avoid the rest of the processing for the action.
```

Comment:
>their execution wraps most of the filter pipeline

What does that mean? Does it mean because they run after auth filter there between their before and after the cover most of the pipeline?